site/app/controllers/admin/GradeableController.php

```diff
@@ -74,4 +99,71 @@ public function process_config_upload() {
         $this->core->redirect($this->core->buildUrl(array('component' => 'admin', 'page' => 'gradeable',
             'action' => 'upload_config')));
     }
+
+    public function rename_config(){
+        $config_file_path = $_POST['curr_config_name'] ?? null;
+        if($config_file_path == null){
+            $this->core->addErrorMessage("Unable to find file");
+
+        } else {
+            $new_name = $_POST['new_config_name'] ?? null;
+            $new_dir = FileUtils::joinPaths(dirname($config_file_path, 1), $new_name);
+            if(rename($config_file_path, $new_dir)){
+                $this->core->addSuccessMessage("Successfully renamed file");
+                $this->core->redirect($this->core->buildUrl(array('component' => 'admin', 'page' => 'gradeable',
+                    'action' => 'upload_config')));
```

Comment:
I think this is redundant because it is called at the end of the function anyway